src/internal_events/common.rs

```diff
@@ -114,3 +110,25 @@ impl InternalEvent for CollectionCompleted {
         histogram!("collect_duration_seconds", self.end - self.start);
     }
 }
+
+#[derive(Debug)]
+pub struct ComponentEventsDropped {
+    pub count: u64,
+    pub intentional: bool,
+    pub reason: &'static str,
+}
+
+impl InternalEvent for ComponentEventsDropped {
+    fn emit(self) {
+        error!(
+            message = "Events dropped.",
+            intentional = self.intentional,
+            reason = self.reason,
+        );
+        counter!(
+            "component_discarded_events_total",
+            self.count,
+            "intentional" => if self.intentional { "true" } else { "false" },
+        );
+    }
```

Comment:
Should `ComponentEventsDropped` increment the deprecated counter! "events_discarded_total" ?